controllers/dspipeline_controller.go

```diff
@@ -189,46 +222,116 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
+	// Initialize conditions
+	var conditions []metav1.Condition
+
+	databaseReady := r.buildCondition(config.DatabaseReady, dspa, "DatabaseReady")
 	err = r.ReconcileDatabase(ctx, dspa, params)
 	if err != nil {
 		return ctrl.Result{}, err
 	}
 
+	if r.isDeploymentAvailable(ctx, dspa, "mariadb") {
+		databaseReady.Status = metav1.ConditionTrue
+	}
+	conditions = append(conditions, databaseReady)
+
+	objectStorageReady := r.buildCondition(config.ObjectStorageReady, dspa, "ObjectStorageReady")
```

Comment:
this can be in a follow up too ... you have a const for `ObjectStorageReady` above

not sure if you should also make consts for `MinimumReplicasAvailable` too